neuro_san/service/http/server/http_sidecar.py

```diff
@@ -107,13 +107,13 @@ def __call__(self, other_server: AgentServer):
                               "Timeout (%d sec) waiting for signal to HTTP server to start",
                               self.TIMEOUT_TO_START_SECONDS)
 
-        app.listen(self.http_port)
-        self.logger.info({}, "HTTP server is running on port %d", self.http_port)
-        self.logger.info({}, "HTTP server is shutting down after %d requests", self.requests_limit)
         # Construct initial "allowed" list of agents:
         # no metadata to use here yet.
         self.update_agents(metadata={})
         self.logger.debug({}, "Serving agents: %s", repr(self.allowed_agents.keys()))
+        app.listen(self.http_port)
+        self.logger.info({}, "HTTP server is running on port %d", self.http_port)
+        self.logger.info({}, "HTTP server is shutting down after %d requests", self.requests_limit)
 
```

Comment:
Start actually listening after all preparations are done.
Log server start at this point.